airflow/jobs/local_task_job.py

```diff
@@ -83,6 +113,14 @@ def signal_handler(signum, frame):
             self.task_runner.terminate()
             self.handle_task_exit(128 + signum)
 
+        def segfault_signal_handler(signum, frame):
+            """Setting sigmentation violation signal handler"""
+            self.log.critical(SIGSEGV_MESSAGE)
+            self.task_runner.terminate()
+            self.handle_task_exit(128 + signum)
+            raise AirflowException("Segmentation Fault detected.")
+
+        signal.signal(signal.SIGSEGV, segfault_signal_handler)
```

Comment:
Third attempt for detect segfault and show info for end users by handling SIGSEGV signal.

@potiuk @uranusjr I'm not sure is this correct way to terminate task runner?

Reply:
I’m not sure either (and not sure even if this would work), but segmentation faults are not the easiest to test, so we might as well just merge this and see how things go…

Reply:
Locally it work fine but only with cases that I know (macOS specific stuff).

Reply:
Agree with @uranusjr . Let's merge and see. When you have SIGSEGV you are screwed anyway, we cannot get it any worse.